### DISCUSSION TITLE
Configurable delays on the tooltip

### DISCUSSION THREAD

**Discussion by jenniferarnesen** [dhis2/notes#230] [Ideas]
I'm finding the tooltip show/hide delays to be a little counterintuitive. Personally, I would prefer a tooltip to wait slightly longer before appearing, and then disappear almost immediately when no longer hovering.

 I realize that this scenario is about a more experienced user of the app, who knows what things are and doesn't need the info in the tooltip. That user will find constantly popping up tooltips to be annoying. But for data-visualizer, won't many users be experienced and generally not need the tooltip?  (am speaking in particular about the data selector - and a tooltip showing on each data item)

**Comment by ghost** [comment]
I'll tag @cooper-joe as he's the authority on our UX/design system.

**Comment by cooper-joe** [comment]
Yes, configurable delays would be a useful addition to the tooltip.  Right now the tooltip is focused on the "guidance" user need. The tooltip like the one @jenniferarnesen describes is more of an advanced "detail exposition" user need. Both are valid use cases. The current tooltip only satisfies the first use case right now, adding configurable delays would allow both use cases (and more).

I've opened a [ticket](https://jira.dhis2.org/browse/LIBS-126) to add this functionality to the Tooltip in `ui`. I've also noted a task for the design team to update the Design System documentation with delay duration guidance once this functionality is available.

Thanks for highlighting this @jenniferarnesen.

**Comment by KaiVandivier** [comment]
PR merged: https://github.com/dhis2/ui/pull/511
